### PR TITLE
Preserve non-decodable %-sequences intact when unquote.

### LINKED ISSUE #517: Only last octet is left in non-UTF-8 %-encoded sequence
```
>>> URL.build(path='/%f0%9f%90', encoded=True).path
'/%90'
```

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Closes #517.

### CODE REVIEW

File: CHANGES/517.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+No longer loss characters when decode incorrect percent-sequences (like ``%e2%82%f8``). All non-decodable percent-sequences are now preserved intact.
```
Comment (webknjaz) [inline CHANGES/517.bugfix:1]:
```suggestion
No longer loose characters when decoding incorrect percent-sequences (like ``%e2%82%f8``). All non-decodable percent-sequences are now preserved.
```

File: setup.cfg [hunk 2]
```diff
@@ -12,7 +12,7 @@ junit_suite_name = yarl_test_suite
 
 
 [flake8]
-ignore = E301,E302,E704,W503,W504,F811
+ignore = E203,E301,E302,E704,W503,W504,F811
```
Comment (webknjaz) [inline setup.cfg:15]: I'd rather inject a few `noqa`s if it's really necessary

Comment (serhiy-storchaka) [reply setup.cfg:15]: Too many `noqa`s should be injected. `blake` produces code incompatible with `flake8`.

**Comment by serhiy-storchaka** [comment]
The downside is the loss of performance.
Before:
```
Cython unquote: 0.305 sec
Python unquote: 2.103 sec
```
After:
```
Cython unquote: 0.753 sec
Python unquote: 2.700 sec
```

**Comment by serhiy-storchaka** [comment]
Fixed the performance regression in Cython:
```
Cython unquote: 0.266 sec
Python unquote: 2.615 sec
```

**Review by asvetlov — APPROVED** [review]
LGTM!

**Pull request merged**